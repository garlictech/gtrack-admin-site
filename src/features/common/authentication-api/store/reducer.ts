// tslint:disable:only-arrow-functions no-small-switch
import { ActionReducer } from '@ngrx/store';

import * as ApiActions from './actions';
import { AuthenticationState } from './state';

export type Action = ApiActions.AllActions;

const initialState: AuthenticationState = {
  auth: {
    token: undefined,
    refreshToken: undefined,
    user: undefined
  },
  loggingIn: true,
  failed: undefined,
  emailSent: false
};

export const reducer: ActionReducer<AuthenticationState> = (state = initialState, action: Action) => {
  switch (action.type) {
    case ApiActions.TWITTER_LOGIN:
    case ApiActions.REQUEST_VERIFY_TOKEN:
    case ApiActions.FACEBOOK_LOGIN:
    case ApiActions.GOOGLE_LOGIN:
    case ApiActions.MAGICLINK_REQUEST_TOKEN:
    case ApiActions.VERIFY:
    case ApiActions.MAGICLINK_LOGIN:
      return { ...state, loggingIn: true, failed: undefined };

    case ApiActions.MAGICLINK_EMAIL_SENT:
      return { ...state, emailSent: true, failed: undefined, loggingIn: false };

    case ApiActions.LOGIN_SUCCESS:
      return { ...{ auth: action.auth }, loggingIn: false, failed: undefined, emailSent: false };

    case ApiActions.USER_CANCELLED:
    case ApiActions.FAILURE_HAPPENED:
      return { ...state, ...{ failed: action.error }, loggingIn: false, emailSent: false };

    case ApiActions.LOGOUT_START:
      return { ...state };

    case ApiActions.LOGOUT_SUCCESS:
      return { ...initialState, loggingIn: false };

    default:
      return state;
  }
};
