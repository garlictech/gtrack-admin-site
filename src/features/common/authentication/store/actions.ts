// tslint:disable:no-property-initializers max-classes-per-file
import { ActivatedRouteSnapshot, RouterStateSnapshot } from '@angular/router';
import { Action } from '@ngrx/store';
import { Auth } from './state';

export enum AuthenticationActionTypes {
  TWITTER_LOGIN = '[Authentication] Login with Twitter',
  REQUEST_VERIFY_TOKEN = '[Authentication] Request Twitter verify token',
  VERIFY = '[Authentication] Verify user with token',
  FACEBOOK_LOGIN = '[Authentication] Login with Facebook',
  GOOGLE_LOGIN = '[Authentication] Login with Google',
  MAGICLINK_LOGIN = '[Authentication] Login with Magic Link',
  MAGICLINK_REQUEST_TOKEN = '[Authentication] Request a Magic Link',
  MAGICLINK_EMAIL_SENT = '[Authentication] Magic Link email sent',
  LOGIN_SUCCESS = '[Authentication] Login Success',
  FAILURE_HAPPENED = '[Authentication] Login/Logout Error',
  LOGOUT_START = '[Authentication] Logout Start',
  LOGOUT_SUCCESS = '[Authentication] Logout Success',
  UNAUTHORIZED = '[Authentication] Unauthorized',
  ROUTE_FORBIDDEN = '[Authentication] Route forbidden',
  USER_CANCELLED = '[Authentication] User cancelled',
  WINDOW_REOPENED = '[Authentication] Window reopened',
  TERMS_ACCEPTED = '[Authentication] Terms accepted',
  LOGIN_REFUSED = '[Authentication] Login refused as term is not accepted',
  SELECT_ROLE = '[Authentication] Auth role selected'
}

export class RequestVerifyToken implements Action {
  readonly type = AuthenticationActionTypes.REQUEST_VERIFY_TOKEN;

  constructor(public email: string) {}
}

export class Verify implements Action {
  readonly type = AuthenticationActionTypes.VERIFY;
  constructor(public token: string, public uid: string) {}
}

export class FacebookLogin implements Action {
  readonly type = AuthenticationActionTypes.FACEBOOK_LOGIN;
  constructor(public roles: Array<string>) {}
}

export class GoogleLogin implements Action {
  readonly type = AuthenticationActionTypes.GOOGLE_LOGIN;
  constructor(public roles: Array<string>) {}
}

export class TwitterLogin implements Action {
  readonly type = AuthenticationActionTypes.TWITTER_LOGIN;
  constructor(public roles: Array<string>) {}
}

export class RequestMagicLinkToken implements Action {
  readonly type = AuthenticationActionTypes.MAGICLINK_REQUEST_TOKEN;
  constructor(public email: string, public language: string, public roles: Array<string>) {}
}

export class MagicLinkLogin implements Action {
  readonly type = AuthenticationActionTypes.MAGICLINK_LOGIN;
  constructor(public token: string, public uid: string, public roles: Array<string>) {}
}

export class MagicLinkEmailSent implements Action {
  readonly type = AuthenticationActionTypes.MAGICLINK_EMAIL_SENT;
}

export class LoginSuccess implements Action {
  readonly type = AuthenticationActionTypes.LOGIN_SUCCESS;
  constructor(public auth: Auth | null) {}
}

export class FailureHappened implements Action {
  readonly type = AuthenticationActionTypes.FAILURE_HAPPENED;
  constructor(public error: any) {}
}

export class UserCancelled implements Action {
  readonly type = AuthenticationActionTypes.USER_CANCELLED;
  constructor(public error: any) {}
}

export class WindowReopened implements Action {
  readonly type = AuthenticationActionTypes.WINDOW_REOPENED;
  // Payload: the error object
  constructor(public error: any) {}
}

export class LogoutStart implements Action {
  readonly type = AuthenticationActionTypes.LOGOUT_START;
}

export class LogoutSuccess implements Action {
  readonly type = AuthenticationActionTypes.LOGOUT_SUCCESS;
}

export class Unauthorized implements Action {
  readonly type = AuthenticationActionTypes.UNAUTHORIZED;
}

export class RouteForbidden implements Action {
  readonly type = AuthenticationActionTypes.ROUTE_FORBIDDEN;

  constructor(public route?: ActivatedRouteSnapshot, public state?: RouterStateSnapshot) {}
}

export class TermsAccepted implements Action {
  readonly type = AuthenticationActionTypes.TERMS_ACCEPTED;
  constructor(public payload: boolean) {}
}

export class LoginRefused implements Action {
  readonly type = AuthenticationActionTypes.LOGIN_REFUSED;
}

export class SelectRole implements Action {
  readonly type = AuthenticationActionTypes.SELECT_ROLE;
  constructor(public payload: string) {}
}

export type AllActions =
  | RequestVerifyToken
  | Verify
  | FacebookLogin
  | GoogleLogin
  | TwitterLogin
  | RequestMagicLinkToken
  | MagicLinkLogin
  | MagicLinkEmailSent
  | LoginSuccess
  | FailureHappened
  | LogoutStart
  | LogoutSuccess
  | Unauthorized
  | RouteForbidden
  | UserCancelled
  | WindowReopened
  | TermsAccepted
  | LoginRefused
  | SelectRole;
