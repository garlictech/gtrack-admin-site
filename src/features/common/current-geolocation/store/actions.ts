// tslint:disable:no-property-initializers max-classes-per-file
import { Action } from '@ngrx/store';
import { GeoPosition } from '../interfaces';

export enum CurrentGeolocationActionTypes {
  CURRENT_LOCATION_OBTAINED = '[Location API] Current Location Obtained',
  START_POSITIONING = '[Location] Start Positioning',
  END_POSITIONING = '[Location] End Positioning'
}

export class CurrentLocationObtained implements Action {
  readonly type = CurrentGeolocationActionTypes.CURRENT_LOCATION_OBTAINED;

  constructor(public location: GeoPosition) {}
}

export class StartPositioning {
  readonly type = CurrentGeolocationActionTypes.START_POSITIONING;
}

export class EndPositioning {
  readonly type = CurrentGeolocationActionTypes.END_POSITIONING;
}

export type Actions = CurrentLocationObtained | StartPositioning | EndPositioning;
