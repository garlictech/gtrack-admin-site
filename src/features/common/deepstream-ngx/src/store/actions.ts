// tslint:disable:no-property-initializers max-classes-per-file
import { IClientData } from '@garlictech/deepstream-rxjs';
import { Action } from '@ngrx/store';

export const DEEPSTREAM_AUTH_SUCCESS = '[Deepstream Authentication] Authentication Success';
export const DEEPSTREAM_LOGIN_SUCCESS = '[Deepstream Authentication] Login Success';
export const DEEPSTREAM_LOGIN = '[Deepstream Authentication] Logging in...';
export const DEEPSTREAM_LOGIN_FAILED = '[Deepstream Authentication] Failed';
export const DEEPSTREAM_LOGOUT_SUCCESS = '[Deepstream Authentication] Logout success';
export const DEEPSTREAM_PERMISSION_RECORD_CHANGED = '[Deepstream] Permission record changed';

export class DeepstreamLoginSuccess implements Action {
  readonly type = DEEPSTREAM_LOGIN_SUCCESS;
  constructor(public clientData: IClientData) {}
}

export class DeepstreamAuthSuccess implements Action {
  readonly type = DEEPSTREAM_AUTH_SUCCESS;
  constructor(public clientData: IClientData) {}
}

export class DeepstreamLogoutSuccess implements Action {
  readonly type = DEEPSTREAM_LOGOUT_SUCCESS;
}

export class DeepstreamLogin implements Action {
  readonly type = DEEPSTREAM_LOGIN;

  // The token
  constructor(public token: string) {}
}
export class DeepstreamLoginFailed implements Action {
  readonly type = DEEPSTREAM_LOGIN_FAILED;
  constructor(public error: any) {}
}

export class DeepstreamPermissionRecordChanged implements Action {
  readonly type = DEEPSTREAM_PERMISSION_RECORD_CHANGED;

  // The new permission record
  constructor(public permissionRecord: any) {}
}

export type AllActions =
  | DeepstreamLogin
  | DeepstreamLoginFailed
  | DeepstreamLoginSuccess
  | DeepstreamLogoutSuccess
  | DeepstreamPermissionRecordChanged
  | DeepstreamAuthSuccess;
