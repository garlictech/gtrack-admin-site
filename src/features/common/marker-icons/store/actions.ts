// tslint:disable:no-property-initializers max-classes-per-file
import { Action } from '@ngrx/store';
import { SvgContent } from './state';

export enum ActionTypes {
  Reset = '[Marker icons] Reset',
  AddSvgIconContents = '[Marker icons] Add SVG icon contents',
  AddSvgMarkerContents = '[Marker icons] Add SVG marker contents'
}

export class Reset implements Action {
  readonly type = ActionTypes.Reset;
}

export class AddSvgIconContents implements Action {
  readonly type = ActionTypes.AddSvgIconContents;
  constructor(public svgContents: Array<SvgContent>) {}
}

export class AddSvgMarkerContents implements Action {
  readonly type = ActionTypes.AddSvgMarkerContents;
  constructor(public svgContents: Array<SvgContent>) {}
}

export type AllActions = Reset | AddSvgIconContents | AddSvgMarkerContents;
