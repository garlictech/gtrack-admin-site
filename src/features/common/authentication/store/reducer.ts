// tslint:disable:only-arrow-functions no-small-switch
import { ActionReducer, combineReducers } from '@ngrx/store';

import { AllActions as Action, AuthenticationActionTypes } from './actions';
import { JwtAuthState, UiState } from './state';

const initialJwtState: JwtAuthState = {
  auth: {
    token: undefined,
    refreshToken: undefined,
    user: undefined
  },
  loggingIn: true,
  failed: undefined,
  emailSent: false
};

const initialUiState: UiState = {
  termsAccepted: false,
  loginRefused: false,
  selectedRole: 'user'
};

export const jwtReducer: ActionReducer<JwtAuthState> = (state = initialJwtState, action: Action) => {
  console.log(action.type);
  switch (action.type) {
    case AuthenticationActionTypes.TWITTER_LOGIN:
    case AuthenticationActionTypes.REQUEST_VERIFY_TOKEN:
    case AuthenticationActionTypes.FACEBOOK_LOGIN:
    case AuthenticationActionTypes.GOOGLE_LOGIN:
    case AuthenticationActionTypes.MAGICLINK_REQUEST_TOKEN:
    case AuthenticationActionTypes.VERIFY:
    case AuthenticationActionTypes.MAGICLINK_LOGIN:
      return { ...state, loggingIn: true, failed: undefined };

    case AuthenticationActionTypes.MAGICLINK_EMAIL_SENT:
      return { ...state, emailSent: true, failed: undefined, loggingIn: false };

    case AuthenticationActionTypes.LOGIN_SUCCESS:
      return { ...{ auth: action.auth }, loggingIn: false, failed: undefined, emailSent: false };

    case AuthenticationActionTypes.USER_CANCELLED:
    case AuthenticationActionTypes.FAILURE_HAPPENED:
      return { ...state, ...{ failed: action.error }, loggingIn: false, emailSent: false };

    case AuthenticationActionTypes.LOGOUT_START:
      return { ...state };

    case AuthenticationActionTypes.LOGOUT_SUCCESS:
      return { ...initialJwtState, loggingIn: false };

    default:
      return state;
  }
};

export const uiErrorReducer: ActionReducer<UiState> = (state = initialUiState, action: Action) => {
  switch (action.type) {
    case AuthenticationActionTypes.TERMS_ACCEPTED:
      return { ...state, termsAccepted: action.payload, loginRefused: false };

    case AuthenticationActionTypes.LOGIN_REFUSED:
      return { ...state, loginRefused: true };

    case AuthenticationActionTypes.SELECT_ROLE:
      return { ...state, selectedRole: action.payload };

    default:
      return state;
  }
};

export const logout = (_reducer: any): any =>
  function(state: any, action: any): any {
    return _reducer(action.type === AuthenticationActionTypes.LOGOUT_SUCCESS ? undefined : state, action);
  };

const metaReducers = [logout];

const _reducers = {
  jwtAuth: jwtReducer,
  uiState: uiErrorReducer
};

export const reducer = combineReducers(_reducers);
export { metaReducers };
