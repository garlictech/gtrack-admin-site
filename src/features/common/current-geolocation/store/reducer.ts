// tslint:disable:only-arrow-functions no-small-switch
import { Actions, CurrentGeolocationActionTypes } from './actions';
import { CurrentGeolocationState } from './state';

export const initialState: CurrentGeolocationState = {
  currentLocation: undefined,
  tracking: false
};

export function reducer(state = initialState, action: Actions): CurrentGeolocationState {
  switch (action.type) {
    case CurrentGeolocationActionTypes.CURRENT_LOCATION_OBTAINED:
      return { ...state, currentLocation: action.location };

    case CurrentGeolocationActionTypes.START_POSITIONING:
      return { ...state, tracking: true };

    case CurrentGeolocationActionTypes.END_POSITIONING:
      return { ...state, tracking: false };

    default:
      return state;
  }
}
