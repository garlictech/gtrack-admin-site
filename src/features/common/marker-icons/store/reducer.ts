// tslint:disable:only-arrow-functions no-small-switch
import { createEntityAdapter, EntityAdapter } from '@ngrx/entity';

import { ActionReducer, ActionReducerMap, combineReducers } from '@ngrx/store';
import { ActionTypes } from './actions';
import { MarkerIconsAction } from './index';
import { State, SvgContent, SvgContentEntityState } from './state';

export const svgIconContentAdapter: EntityAdapter<SvgContent> = createEntityAdapter<SvgContent>();
export const svgIconContentInitialState = svgIconContentAdapter.getInitialState();

export const svgMarkerContentAdapter: EntityAdapter<SvgContent> = createEntityAdapter<SvgContent>();
export const svgMarkerContentInitialState = svgMarkerContentAdapter.getInitialState();

export const initialState: State = {
  icons: undefined,
  markers: undefined
};

export function svgIconsReducer(
  state: SvgContentEntityState = svgIconContentInitialState,
  action: MarkerIconsAction
): SvgContentEntityState {
  switch (action.type) {
    case ActionTypes.Reset:
      return { ...svgIconContentInitialState };

    case ActionTypes.AddSvgIconContents: {
      return svgIconContentAdapter.addAll(action.svgContents, state);
    }

    default:
      return state;
  }
}

export function svgMarkerReducer(
  state: SvgContentEntityState = svgMarkerContentInitialState,
  action: MarkerIconsAction
): SvgContentEntityState {
  switch (action.type) {
    case ActionTypes.Reset:
      return { ...svgMarkerContentInitialState };

    case ActionTypes.AddSvgMarkerContents: {
      return svgIconContentAdapter.addAll(action.svgContents, state);
    }

    default:
      return state;
  }
}

const reducerMap: ActionReducerMap<State> = {
  icons: svgIconsReducer,
  markers: svgMarkerReducer
};

const reducer: ActionReducer<State> = combineReducers(reducerMap);

export function markerIconsReducer(state: any, action: any): State {
  return reducer(state, action);
}
