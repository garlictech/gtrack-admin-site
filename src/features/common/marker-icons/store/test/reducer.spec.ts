import * as _ from 'lodash';

import * as fromActions from '../actions';
import { markerIconsReducer } from '../reducer';
import { createFeatureState, createSvgContent, createSvgContentEntityState } from './utils';

const contents = [createSvgContent('atm'), createSvgContent('bank')];

const testCases = [
  {
    actionName: 'Reset',
    action: new fromActions.Reset(),
    initialState: createFeatureState({
      markers: createSvgContentEntityState(contents),
      icons: createSvgContentEntityState(contents)
    })
  },
  {
    actionName: 'AddSvgIconContents',
    action: new fromActions.AddSvgIconContents(contents),
    initialState: createFeatureState()
  },
  {
    actionName: 'AddSvgMarkerContents',
    action: new fromActions.AddSvgMarkerContents(contents),
    initialState: createFeatureState()
  }
];

testCases.forEach(({ actionName, action, initialState }) => {
  test(`${actionName} with initial state ${JSON.stringify(initialState)}`, () => {
    expect(markerIconsReducer(initialState, action)).toMatchSnapshot();
  });
});
