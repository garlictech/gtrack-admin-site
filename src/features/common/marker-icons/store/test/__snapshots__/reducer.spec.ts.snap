// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSvgIconContents with initial state {} 1`] = `
Object {
  "circles": Object {
    "entities": Object {},
    "ids": Array [],
  },
  "icons": Object {
    "entities": Object {
      "atm": Object {
        "content": "<svg><path id=\\"icon\\" fill=\\"#7C7CF0\\"></path></svg>",
        "id": "atm",
      },
      "bank": Object {
        "content": "<svg><path id=\\"icon\\" fill=\\"#7C7CF0\\"></path></svg>",
        "id": "bank",
      },
    },
    "ids": Array [
      "atm",
      "bank",
    ],
  },
  "markers": Object {
    "entities": Object {},
    "ids": Array [],
  },
}
`;

exports[`AddSvgMarkerContents with initial state {} 1`] = `
Object {
  "circles": Object {
    "entities": Object {},
    "ids": Array [],
  },
  "icons": Object {
    "entities": Object {},
    "ids": Array [],
  },
  "markers": Object {
    "entities": Object {
      "atm": Object {
        "content": "<svg><path id=\\"icon\\" fill=\\"#7C7CF0\\"></path></svg>",
        "id": "atm",
      },
      "bank": Object {
        "content": "<svg><path id=\\"icon\\" fill=\\"#7C7CF0\\"></path></svg>",
        "id": "bank",
      },
    },
    "ids": Array [
      "atm",
      "bank",
    ],
  },
}
`;

exports[`Reset with initial state {"markers":{"ids":["atm","bank"],"entities":{"atm":{"id":"atm","content":"<svg><path id=\\"icon\\" fill=\\"#7C7CF0\\"></path></svg>"},"bank":{"id":"bank","content":"<svg><path id=\\"icon\\" fill=\\"#7C7CF0\\"></path></svg>"}}},"icons":{"ids":["atm","bank"],"entities":{"atm":{"id":"atm","content":"<svg><path id=\\"icon\\" fill=\\"#7C7CF0\\"></path></svg>"},"bank":{"id":"bank","content":"<svg><path id=\\"icon\\" fill=\\"#7C7CF0\\"></path></svg>"}}}} 1`] = `
Object {
  "circles": Object {
    "entities": Object {},
    "ids": Array [],
  },
  "icons": Object {
    "entities": Object {},
    "ids": Array [],
  },
  "markers": Object {
    "entities": Object {},
    "ids": Array [],
  },
}
`;
